### PR TITLE
Hy/Test download pdf with form fields

### PR DESCRIPTION
**Pull request by sv-hyacoub**
# Description

Verify that a pdf with form fields can be filled and downloaded correctly.

## Bugzilla bug ID

**[Testrail](https://mozilla.testrail.io/index.php?/cases/view/1020326)**
**[Link](https://bugzilla.mozilla.org/show_bug.cgi?id=1910713)**

## Type of change

Please delete options that are not relevant.

- [x] New Test

# How does this resolve / make progress on that bug?

Completed

# Screenshots / Explanations

N/A

# Comments / Concerns

N/A

### CODE REVIEW

File: modules/data/generic_pdf.components.json [hunk 1]
```diff
@@ -37,5 +37,17 @@
         "selectorData": "download",
         "strategy": "id",
         "groups": []
+    },
+
+    "first-name-field": {
+        "selectorData": "pdfjs_internal_id_638R",
```
Comment (ben-c-at-moz) [inline modules/data/generic_pdf.components.json:43]: I don't think I trust the PDFjs internal ids, I would change to `input[name='First Name Given Name']` with strategy `css`

File: modules/data/generic_pdf.components.json [hunk 2]
```diff
@@ -37,5 +37,17 @@
...
+        "strategy": "id",
+        "groups": []
+    },
+
+    "edited-field": {
+        "selectorData": "[id='pdfjs_internal_id_638R'][value='Mark']",
```
Comment (ben-c-at-moz) [inline modules/data/generic_pdf.components.json:49]: I recommend making a change to this selector as suggested for the previous

**Review by ben-c-at-moz — CHANGES_REQUESTED** [review]
Just a couple of quick things...

File: modules/data/generic_pdf.components.json (hunk 2 above)
Comment (sv-hyacoub) [reply modules/data/generic_pdf.components.json]: Made the changes. Thanks.

**Review by ben-c-at-moz — APPROVED** [review]

**Pull request merged**